app/src/main/java/com/andrew/tasky/agenda/domain/models/AgendaItem.kt

```diff
@@ -2,27 +2,44 @@ package com.andrew.tasky.agenda.domain.models
 
 import androidx.room.Entity
 import androidx.room.PrimaryKey
-import com.andrew.tasky.agenda.util.AgendaItemType
 import com.andrew.tasky.agenda.util.ReminderTime
 import java.io.Serializable
 import java.time.LocalDateTime
 
-@Entity(
-    tableName = "agendaItems"
-)
+sealed interface AgendaItem : Serializable {
 
-data class AgendaItem(
-    @PrimaryKey(autoGenerate = true) var id: Int? = null,
-    val apiId: String,
-    val type: AgendaItemType,
-    var isDone: Boolean,
-    val title: String,
-    val description: String,
-    val startDateAndTime: LocalDateTime,
-    val endDateAndTime: LocalDateTime? = null,
-    val reminderTime: ReminderTime,
-    val photos: List<EventPhoto>? = emptyList(),
-    val isAttendee: Boolean? = null,
-    val attendees: List<Attendee>? = emptyList(),
-    val host: String? = null
-) : Serializable
+    @Entity
+    data class Event(
```

Comment:
Here you're coupling domain with data related dependencies (in this case Room and its annotations). The domain layer should be isolated from all kinds of APIs and databases and it should never know where data comes from. That decision lies in the data layer.
So in your case, these entities would need to be moved to your data layer. Then you will have a mapper function which maps these entities to a corresponding domain model which doesn't have any Room or JSON annotations. The UI then uses these domain models to display stuff.